### PR TITLE
particl-qt: init at 0.18.1.4

### PR DESCRIPTION
**Pull request by demyanrogozhin**
###### Motivation for this change

Particl QT wallet that can be used to activate cold staking with Ladger hardware wallet.
Works Linux only so far.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by joachifm** [comment]
@GrahamcOfBorg build particl-qt

**Comment by xeji** [comment]
Please rebase and resolve merge conflict. You can also try `enableParallelBuilding = true;` here to speed up the build. Also I think it would make sense to use the same version of `boost` as for `particl-core` (if that works).

**Comment by demyanrogozhin** [comment]
I rebase and did same with boost. Also check UI tests - working. I guess `enableParallelBuilding` was inherited from particl-core into this derivation @xeji

**Comment by demyanrogozhin** [comment]
Im confused, this built well on my laptop. Also I already saw this error when I try to build .17-Alfa, but back then I decide its due to unstable version.
Now I have to find why build is not reproducible. Its ether parallelism or libhid.

**Comment by xeji** [comment]
Let's start the build again and see if anything changes. 
@GrahamcOfBorg particl-qt

**Comment by xeji** [comment]
@GrahamcOfBorg build particl-qt

**Comment by xeji** [comment]
Since the build for x86_64-linux succeeded now I would assume the parallel build is non-deterministic (some internal dependencies not properly specified). The obvious solution is to disable parallel building.

**Comment by demyanrogozhin** [comment]
Thanks @xeji ! 

Disabled parallel building option for particl-qt and rebased to latest master.

It appears that C headers that should be generated by protobuf before compilation of usbdevice/ folder (needed for trezor hardware wallet support) occasionally was not generated during parallel build. Something for Particl team to look at.

**Comment by xeji** [comment]
> Something for Particl team to look at.

I agree. It would be great if you could report that upstream.
So let's try the build again. It may time out without parallel building, we'll see...

@GrahamcOfBorg build particl-qt

File: pkgs/applications/altcoins/particl/particl-qt.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  configureFlags = oldAttrs.configureFlags ++ [
+    "--with-gui=qt5"
+    "--enable-usbdevice=yes"
+    "--with-qt-bindir=${qtbase}/bin:${qttools}/bin:${qtbase.dev}/bin:${qttools.dev}/bin"
+  ] ++ stdenv.lib.optionals (doCheck) [
+    "--enable-gui-tests=yes"
+  ];
+
+  # Parallel building become non-deterministic for particl-qt (see nixpkgs pr #46760)
+  enableParallelBuiling = false;
```
Comment (xeji) [inline pkgs/applications/altcoins/particl/particl-qt.nix]: because of the typo here, parallel building was NOT disabled...

Comment (xeji) [reply pkgs/applications/altcoins/particl/particl-qt.nix]: Thanks!

**Comment by xeji** [comment]
@GrahamcOfBorg build particl-qt

**Comment by demyanrogozhin** [comment]
Same error. Now I'm not sure if parallel build has to do something with it.
At this point I would rather wait for next official stable release with Makefile fix.

**Comment by xeji** [comment]
I agree. Let's mark this as WIP and wait.

**Comment by mmahut** [comment]
Are there any updates on this pull request, please?

**Comment by demyanrogozhin** [comment]
@mmahut pull request updated with actual version of source.

**Comment by demyanrogozhin** [comment]
Closing, because mainline Particle-Core updated to v0.19 this PR adds outdated version.

**Pull request closed without merging**